From 93d73f386a5bac3068ed7843fa96b30f8274c912 Mon Sep 17 00:00:00 2001
From: "Steven Rostedt (Red Hat)" <rostedt@goodmis.org>
Date: Fri, 16 Sep 2016 11:15:27 -0400
Subject: [PATCH 102/134] trace-cmd: Make debug a global variable

If one application enables debugging, have all files see it.

Signed-off-by: Steven Rostedt <rostedt@goodmis.org>
---
 trace-cmd.c    | 2 ++
 trace-listen.c | 2 --
 trace-local.h  | 2 ++
 trace-read.c   | 1 -
 trace-record.c | 1 -
 5 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/trace-cmd.c b/trace-cmd.c
index 1a62faaf71ca..bbb5b10d9fa5 100644
--- a/trace-cmd.c
+++ b/trace-cmd.c
@@ -29,6 +29,8 @@
 int silence_warnings;
 int show_status;
 
+int debug;
+
 void warning(const char *fmt, ...)
 {
 	va_list ap;
diff --git a/trace-listen.c b/trace-listen.c
index 1890b6dfcd57..86f6d966f6c1 100644
--- a/trace-listen.c
+++ b/trace-listen.c
@@ -49,8 +49,6 @@ static char *output_file;
 
 static FILE *logfp;
 
-static int debug;
-
 static int backlog = 5;
 
 static int proto_ver;
diff --git a/trace-local.h b/trace-local.h
index 0ac39bfa3f40..a97dca6ebded 100644
--- a/trace-local.h
+++ b/trace-local.h
@@ -25,6 +25,8 @@
 
 #include "trace-cmd.h"
 
+extern int debug;
+
 /* fix stupid glib guint64 typecasts and printf formats */
 typedef unsigned long long u64;
 
diff --git a/trace-read.c b/trace-read.c
index 79519bd67700..40055a3f3c9b 100644
--- a/trace-read.c
+++ b/trace-read.c
@@ -107,7 +107,6 @@ static int stacktrace_id;
 static int profile;
 
 static int buffer_breaks = 0;
-static int debug = 0;
 
 static int no_irqs;
 static int no_softirqs;
diff --git a/trace-record.c b/trace-record.c
index b0429936771b..61b5e61cb321 100644
--- a/trace-record.c
+++ b/trace-record.c
@@ -4169,7 +4169,6 @@ void trace_record (int argc, char **argv)
 	int topt = 0;
 	int do_child = 0;
 	int data_flags = 0;
-	int debug = 0;
 
 	int c;
 
-- 
2.5.5

